sdk/translation/Azure.AI.Translation.Document/tests/GetAllDocumentsFilterTests.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.AI.Translation.Document.Tests
+{
+    public partial class DocumentTranslationClientLiveTests
+    {
+        private async Task<DocumentTranslationOperation> CreateSingleTranslationJobAsync(DocumentTranslationClient client, int docsCount = 1)
+        {
+            var testDocs = CreateDummyTestDocuments(count: docsCount);
+            var sourceContainer = await CreateSourceContainerAsync(testDocs);
+
+            var targetContainer = await CreateTargetContainerAsync();
+            var input = new DocumentTranslationInput(sourceContainer, targetContainer, "fr");
+            var translationOp = await client.StartTranslationAsync(input);
+
+            return translationOp;
+        }
+
+        [RecordedTest]
+        public async Task GetAllDocumentStatusesFilterByStatusTest()
+        {
+            // create client
+            var client = GetClient();
+
+            // create translation job
+            var operation = await CreateSingleTranslationJobAsync(client, docsCount: 3);
+            await operation.WaitForCompletionAsync();
+
+            // list docs
+            var filter = new DocumentFilter(
+                statuses: new List<DocumentTranslationStatus>()
+                {
+                    DocumentTranslationStatus.Succeeded
+                }
+            );
+            var result = operation.GetAllDocumentStatuses(filter: filter);
+
+            // assert.
+            Assert.That(result.All(d => d.Status == DocumentTranslationStatus.Succeeded));
+        }
+
+        [RecordedTest]
+        public async Task GetAllDocumentStatusesFilterByIdsTest()
+        {
+            // create client
+            var client = GetClient();
+
+            // create translation job
+            var operation = await CreateSingleTranslationJobAsync(client, docsCount: 3);
+            await operation.WaitForCompletionAsync();
+            var testIds = operation.GetAllDocumentStatuses().Select(d => d.Id).ToList().GetRange(0, 1);
+
+            // list docs
+            var filter = new DocumentFilter(
+                ids: testIds
+            );
+            var result = operation.GetAllDocumentStatuses(filter: filter);
+
+            // assert
+            Assert.That(result.All(d => testIds.Contains(d.Id)));
+        }
+
+        [Ignore("no way to test this filter")]
+        [RecordedTest]
+        public void GetAllDocumentStatusesFilterByCreatedAfter()
+        {
+        }
+
+        [Ignore("no way to test this filter")]
+        [RecordedTest]
+        public void GetAllDocumentStatusesFilterByCreatedBefore()
+        {
```

Comment:
you could create mock tests for this and just verify that in the SDK we are sending the correct parameter name and date format.
An example of a mock test in TA is: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/textanalytics/Azure.AI.TextAnalytics/tests/AnalyzeOperationMockTests.cs#L42

Reply:
tried out the second approach
but the sorting option apparently doesn't work
returned docs count = 0

Reply:
can you create an issue for this so after this PR we can add these tests?